bin/install-pr

```diff
@@ -9,63 +9,193 @@
 
 set -eo pipefail
 
+cleanup() {
```

Comment:
this highly condition cleanup function is difficult to follow and feels like it will be annoying to maintain.  We also may not want to delete the tmp directory in the case of failures, so that we can debug.  Can we just put the messaging inline where appropriate?  We can check exit codes as we go if necessary.

Reply:
It's not the prettiest, but it also does a good job at explaining if the script fails where it failed and what to do.

I can remove it, but it also then becomes harder to figure out where you have to manually continue the script.

Is there anything I can do that would give you less concern? In testing this, it was helpful knowing that the CLI was installed but `check` failed for some (usually Kubernetes flakiness) reason. Running `linkerd check` again after waiting a few moments would succeed.

Reply:
My point is that we can print appropriate output as we go rather than tracking state and then deferring into this cleanup function.  I'll also point out that this is a script intended for developers so that shifts the priorities towards maintainability and away from fool-proof-ness, somewhat.  Personally, when I see the failure output from `linkerd check`, it's clear to me what happened (linkerd check failed) without a separate message explaining it.

Reply:
Yep that makes sense. It's much more maintainable without `cleanup()` and the intermediate variables it was using.